watertap/core/zero_order_base.py

```diff
@@ -0,0 +1,277 @@
+            raise ConfigurationError(
+                f"{self.name} configured with invalid property package. "
+                f"Zero-order models only support `H2O` as a solvent and all "
+                f"other species as Solutes.")
+
+    def initialize(self, state_args=None, outlvl=idaeslog.NOTSET,
+                   solver=None, optarg=None):
+        '''
+        Placeholder initialization routine, raises NotImplementedError
+        '''
+        if self._initialize is None or not callable(self._initialize):
+            raise NotImplementedError()
+        else:
+            self._initialize(self, state_args=None, outlvl=idaeslog.NOTSET,
+                             solver=None, optarg=None)
+
+    def calculate_scaling_factors(self):
+        '''
+        Placeholder scaling routine, should be overloaded by derived classes
+        '''
+        super().calculate_scaling_factors()
+
+        if callable(self._scaling):
+            self._scaling(self)
+
+    def load_parameters_from_database(self, use_default_removal=False):
+        """
+        Method to load parameters for from database.
+
+        Args:
+            use_default_removal - (optional) indicate whether to use defined
+                                  default removal fraction if no specific value
+                                  defined in database
+
+        Returns:
+            None
+        """
+        # Get parameter dict from database
+        if self._tech_type is None:
+            raise NotImplementedError(
+                f"{self.name} derived zero order unit model has not "
+                f"implemented the _tech_type attribute. This is required "
+                f"to identify the database file to load parameters from.")
+
+        # Get parameter dict from database
+        pdict = self.config.database.get_unit_operation_parameters(
+            self._tech_type, subtype=self.config.process_subtype)
+
+        if self._has_recovery_removal:
+            self.set_recovery_and_removal(pdict, use_default_removal)
+
+        for v in self._fixed_perf_vars:
+            self.set_param_from_data(v, pdict)
+
+    def set_recovery_and_removal(self, data, use_default_removal=False):
+        """
+        Common utiltiy method for setting values of recovery and removal
+        fractions.
+
+        Args:
+            data - dict of parameter values to use when fixing variables
+            use_default_removal - (optional) indicate whether to use defined
+                                  default removal fraction if no specific value
+                                  defined in database
+
+        Returns:
+            None
+        """
+        self.set_param_from_data(self.recovery_vol, data)
+
+        for t, j in self.removal_frac_mass_solute:
+            self.set_param_from_data(
+                self.removal_frac_mass_solute[t, j],
+                data,
+                index=j,
+                use_default_removal=use_default_removal)
```

Comment:
I was imagining that we could have these `removal_frac_mass_solutes` also include water. That way the models could be full specified for the mass balance by either having a `removal_frac_mass_comp` for all components (including water), or for all except for water and then a `recovery_vol`. This would let us get away from setting `recovery_vol` for those units that really just remove components and leave the water alone.

This could be saved for another PR, but I'd like your thoughts.

Reply:
I would suggest not doing that, as it leaves an opening for over-specifying the problem (i.e. fixing both the removal and recovery of water). For cases like you mentioned, there is nothing stopping us from having a different subtype of the material balances where water is unaffected (e.g. an optional argument that says that water recovery is assumed to be 100% so don't include it).

This also touches on the numerical issues associated with stream with zero water flow.